internal/core/items.rs

```diff
@@ -200,6 +200,18 @@ impl ItemWeak {
     pub fn upgrade(&self) -> Option<ItemRc> {
         self.component.upgrade().map(|c| ItemRc::new(c, self.index))
     }
+
+    pub fn component(&self) -> ComponentWeak {
+        self.component.clone()
+    }
+
+    pub fn index(&self) -> usize {
```

Comment:
What's the use case of this function?  This seems useless to have an index without the component.
Perhaps this should be documented what that, at least

Reply:
See above.